src/linkify-element.js

```diff
@@ -32,8 +32,11 @@ function tokensToNodes(tokens, opts, doc) {
 
 	for (let i = 0; i < tokens.length; i++) {
 		let token = tokens[i];
+		let validated = options.resolve(opts.validate,
```

Comment:
I would add a short circuit guard here, so that we don't check on every token (i.e., generic text tokens)

```js
let validated = token.isLink && options.resolve(opts.validate ...
```